partcad_skyline/skyline.py

```
@@ -0,0 +1,146 @@
+            data=json.dumps({"data": commits}),
+            timeout=10,
+        )
+        if response.status_code == 200:
+            logging.info("Commits successfully sent to the API.")
+            logging.info(f"API response: {response.json()}")
+        else:
+            raise click.UsageError(
+                f"Failed to send commits to the API. Status code: {response.status_code}",
+            )
+    except requests.RequestException as e:
+        raise click.UsageError(f"Error sending commits to the API: {e}")
```

Ruff (0.8.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)